Java/core/src/main/java/com/amazon/randomcutforest/store/SmallNodeStore.java

```diff
@@ -153,15 +168,71 @@ public int getMass(int index) {
         return mass[index];
     }
 
+    @Override
+    public void setMass(int index, int newMass) {
+        mass[index] = (short) newMass;
+    }
+
+    @Override
     public void increaseMassOfAncestorsAndItself(int index) {
         while (index != NULL) {
             ++mass[index];
             index = parentIndex[index];
         }
     }
 
+    @Override
+    public void decreaseMassOfAncestorsAndItself(int index) {
+        while (index != (short) NULL) {
+            --mass[index];
+            index = parentIndex[index];
+        }
+    }
+
+    @Override
     public int getSibling(int parent, int node) {
         return leftIndex[parent] == (short) node ? rightIndex[parent] : leftIndex[parent];
     }
 
+    @Override
+    public Map<Integer, Integer> getLeavesAndParents() {
+        HashMap<Integer, Integer> newMap = new HashMap<>();
+        for (int i = 0; i < capacity; i++) {
+            if (occupied.get(i) && leftIndex[i] >= capacity) {
+                newMap.put((int) leftIndex[i], i);
```

Comment:
Did Java complain about this? I thought it would promote a short to an int automatically.

Reply:
Yes, it suggests the cast.